pkg/broker/dynamicplans/plan.go

```diff
@@ -74,3 +76,29 @@ func (p Plan) String() string {
 
 	return string(s)
 }
+
+func (p *Plan) UnmarshalJSON(data []byte) error {
+	type plan Plan
+
+	var pl plan
+	err := json.Unmarshal(data, &pl)
+
+	// try to fix the plan in case of an error
+	if err != nil && pl.APIKey != nil {
```

Comment:
Are we ever writing this fixed data back to Realm? 
If not, then this repair code will always be in use. That's probably good for this release, but we should fix it in a future release

Reply:
I think it is being rewritten as it is.

Reply:
Ok - could you confirm please? It'd be good to know before we ship